custom_metrics/a11y.js

```diff
@@ -0,0 +1,243 @@
+    for (const prop of props) {
+      error_obj[prop] = error[prop].toString();
+    }
+
+    return error_obj;
+  }
+
+  return error;
+}
+
+function captureAndLogError(fun) {
+  try {
+    return fun();
+  } catch (error) {
+    return {
+      __error: getJsonReadyError(error),
+    }
+  }
+}
+
+return JSON.stringify({
+  tables: captureAndLogError(() => {
+    const tables = document.querySelectorAll('table');
+    const tables_with_caption = document.querySelectorAll('table caption');
+    const tabels_with_presentational = document.querySelectorAll('table[role="presentation" i]');
+
+    return {
+      total: tables.length,
+      total_with_caption: tables_with_caption.length,
+      total_with_presentational: tabels_with_presentational.length,
+    };
+  }),
+  file_extension_alts: captureAndLogError(() => {
+    const elements_with_alt_text = [...document.querySelectorAll('img[alt]')];
+    const image_file_extensions = [
+      // ICO
+      'ico', 'cur',
+      // SVG
+      'svg', 'svgz',
+      // JPG
+      'jpg', 'jpeg', 'jpe', 'jif', 'jfif', 'jfi', 'pjpeg', 'pjp',
+      // APNG
+      'apng',
+      // PNG
+      'png',
+      // GIF
+      'gif',
+      // WEBP
+      'webp',
+      // TIFF
+      'tiff', 'tif',
+      // BMP
+      'bmp', 'dib',
+      // JPEG 2000
+      'jp2', 'j2k', 'jpf', 'jpx', 'jpm', 'mj2',
+      // HEIF
+      'heif', 'heic',
+    ];
+    const extension_regex = new RegExp(`\.(${image_file_extensions.join('|')})$`, 'i');
+
+    let total_elements_with_non_empty_alt = 0;
+    let total_with_file_extension = 0;
+    const file_extension_collector = {};
+    for (const element of elements_with_alt_text) {
+      const alt = element.alt.trim().replace(/\s+/g, ' ').toLocaleLowerCase();
+      if (alt.length <= 0) {
+        continue;
+      }
+
+      const matches = alt.match(extension_regex);
+      if (matches && matches[1]) {
+        total_with_file_extension++;
+        incrementCollectorKey(file_extension_collector, matches[1]);
+      }
+    }
+
+    return {
+      total_elements_with_non_empty_alt,
+      total_with_file_extension,
+      file_extensions: file_extension_collector,
+    };
+  }),
+  title_and_alt: captureAndLogError(() => {
+    const has_both_title_and_alt = document.querySelectorAll('*[title][alt]');
+
+    let total_alt_same_as_title = 0;
+    for (const element of has_both_title_and_alt) {
+      const title = (element.getAttribute('title') || '').replace(/\s+/g, ' ').trim().toLocaleLowerCase();
+      const alt = (element.getAttribute('alt') || '').replace(/\s+/g, ' ').trim().toLocaleLowerCase();
+      if (title === alt) {
+        total_alt_same_as_title++;
+      }
+    }
+
+    return {
+      total_alt: document.querySelectorAll('*[alt]').length,
+      total_title: document.querySelectorAll('*[title]').length,
+      total_both: has_both_title_and_alt.length,
+      total_alt_same_as_title,
+    };
+  }),
+  no_destination_anchors: captureAndLogError(() => {
```

Comment:
Is there anything this function does that isn't already covered by https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/wpt_bodies.js#L144 and https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/almanac.js#L283?

Could we remove if duplicate?

Reply:
They are detecting empty links as same page links... which is honestly probably better than what I'm doing here.

Deleting this as its a dupe.